lib/backend/sqlite.c

```diff
@@ -584,11 +558,8 @@ static rpmRC sqlite_idxdbIter(dbiIndex dbi, dbiCursor dbc, const char *keyp, siz
 	    dbc->key = sqlite3_column_blob(dbc->stmt, 0);
 	}
 	dbc->keylen = sqlite3_column_bytes(dbc->stmt, 0);
-	if (set) {
-	    dbiCursor kc = dbiCursorInit(dbi, 0);
-	    rc = sqlite_idxdbByKey(dbi, kc, dbc->key, dbc->keylen, set);
-	    dbiCursorFree(dbi, kc);
-	}
+	if (set)
+	    rc = sqlite_idxdbByKey(dbi, dbc->subc, dbc->key, dbc->keylen, set);
 	rc = RPMRC_OK;
```

Comment:
Is this statement supposed to be here? It essentially clobbers the result from the line above...

Reply:
It's a bit fishy alright, but it's not being introduced in this patch.

IIRC it's there for a reason though, and with this patch all the more so.